From d147841ceabd39a32ec9211bca6cf13f5c86845f Mon Sep 17 00:00:00 2001
From: "Ruiqiang.Hao" <Ruiqiang.Hao@windriver.com>
Date: Thu, 17 Oct 2019 02:16:58 +0000
Subject: [PATCH 372/386] cpu errata: adapt gpio thunderx and cpu errata
 interface to kernel 4.18.45

cpu errata interface update.
change interface and struct members to fix compile errors.

Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/kernel/cpu_errata.c | 49 ++++++++++++++++------------------
 drivers/gpio/gpio-thunderx.c   |  2 +-
 2 files changed, 24 insertions(+), 27 deletions(-)

diff --git a/arch/arm64/kernel/cpu_errata.c b/arch/arm64/kernel/cpu_errata.c
index 2dd38587efbb..60299c17d86e 100644
--- a/arch/arm64/kernel/cpu_errata.c
+++ b/arch/arm64/kernel/cpu_errata.c
@@ -89,7 +89,7 @@ cpu_enable_trap_ctr_access(const struct arm64_cpu_capabilities *__unused)
 atomic_t arm64_el2_vector_last_slot = ATOMIC_INIT(-1);
 
 #ifdef CONFIG_CAVIUM_ERRATUM_36890
-static int cpu_enable_trap_zva_access(void *__unused)
+static void cpu_enable_trap_zva_access(const struct arm64_cpu_capabilities *__unused)
 {
 	/*
 	 * Clear SCTLR_EL2.DZE or SCTLR_EL1.DZE depending
@@ -528,6 +528,17 @@ static const struct midr_range arm64_harden_el2_vectors[] = {
 
 #endif
 
+#ifdef CONFIG_CAVIUM_ERRATUM_36890
+
+static const struct midr_range arm64_cavium_erratum_36890[] = {
+	MIDR_ALL_VERSIONS(MIDR_THUNDERX),
+	MIDR_ALL_VERSIONS(MIDR_OCTEON_T81),
+	MIDR_ALL_VERSIONS(MIDR_OCTEON_T83),
+	{},
+};
+
+#endif
+
 const struct arm64_cpu_capabilities arm64_errata[] = {
 #if	defined(CONFIG_ARM64_ERRATUM_826319) || \
 	defined(CONFIG_ARM64_ERRATUM_827319) || \
@@ -634,43 +645,29 @@ const struct arm64_cpu_capabilities arm64_errata[] = {
 #endif
 #ifdef CONFIG_CAVIUM_ERRATUM_36890
 	{
-		/* Cavium ThunderX, T88 all passes */
-		.desc = "Cavium erratum 36890",
-		.capability = ARM64_WORKAROUND_CAVIUM_36890,
-		MIDR_ALL_VERSIONS(MIDR_THUNDERX),
-		.enable = cpu_enable_trap_zva_access,
-	},
-	{
-		/* Cavium ThunderX, T81 all passes */
-		.desc = "Cavium erratum 36890",
-		.capability = ARM64_WORKAROUND_CAVIUM_36890,
-		MIDR_ALL_VERSIONS(MIDR_OCTEON_T81),
-		.enable = cpu_enable_trap_zva_access,
-	},
-	{
-		/* Cavium ThunderX, T83 all passes */
+		/* Cavium ThunderX, T88 / T81 / T83 all passes */
 		.desc = "Cavium erratum 36890",
 		.capability = ARM64_WORKAROUND_CAVIUM_36890,
-		MIDR_ALL_VERSIONS(MIDR_OCTEON_T83),
-		.enable = cpu_enable_trap_zva_access,
+		ERRATA_MIDR_RANGE_LIST(arm64_cavium_erratum_36890),
+		.cpu_enable = cpu_enable_trap_zva_access,
 	},
 	{
 		/* Marvell OcteonTX 2, 96xx pass A0, A1, and B0 */
 		.desc = "Cavium erratum 36890",
 		.capability = ARM64_WORKAROUND_CAVIUM_36890,
-		MIDR_RANGE(MIDR_MRVL_OCTEONTX2_96XX,
-				MIDR_CPU_VAR_REV(0, 0),
-				MIDR_CPU_VAR_REV(1, 0)),
-		.enable = cpu_enable_trap_zva_access,
+		ERRATA_MIDR_RANGE(MIDR_MRVL_OCTEONTX2_96XX,
+				0, 0,
+				1, 0),
+		.cpu_enable = cpu_enable_trap_zva_access,
 	},
 	{
 		/* Marvell OcteonTX 2, 95 pass A0/A1 */
 		.desc = "Cavium erratum 36890",
 		.capability = ARM64_WORKAROUND_CAVIUM_36890,
-		MIDR_RANGE(MIDR_MRVL_OCTEONTX2_95XX,
-				MIDR_CPU_VAR_REV(0, 0),
-				MIDR_CPU_VAR_REV(0, 1)),
-		.enable = cpu_enable_trap_zva_access,
+		ERRATA_MIDR_RANGE(MIDR_MRVL_OCTEONTX2_95XX,
+				0, 0,
+				0, 1),
+		.cpu_enable = cpu_enable_trap_zva_access,
 	},
 #endif
 	{
diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index 3a1fced5e328..19968e00fb48 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -615,7 +615,7 @@ static void thunderx_gpio_spi_irq_handler(struct irq_desc *desc)
 	for (line = 0; line < chip->ngpio; line++) {
 		if (readq(gpio->register_base + intr_reg(line)) &
 		    GPIO_INTR_INTR) {
-			generic_handle_irq(irq_find_mapping(chip->irqdomain,
+			generic_handle_irq(irq_find_mapping(gpio->irqd,
 							    line));
 			writeq(GPIO_INTR_INTR,
 			       gpio->register_base + intr_reg(line));
-- 
2.17.1

